### PR TITLE
Fix: Allow smartSplit() to handle String of chars not just a single char

### PR DESCRIPTION
**Pull request by omkarnix**
This commit adds the support of String separator in function smartSplit. Previously only char separator is supported. Now smartSplit can support String as well as single char as separator. Changed to existing implementation of overloaded constructor of CSVParser which takes string separator.

### CODE REVIEW

**Comment by wetneb** [comment]
Hi @omkarnix, thanks a lot for this pull request! As discussed in the issue it would be great to have a few tests for this (as you have noticed the test coverage is really poor at the moment so we are trying to improve that for all new changes). Let me know if you need any help.

**Comment by omkarnix** [comment]
@wetneb, Sure Thanks for reviewing. Actually, I realized that low test coverage after creating PR and started the discussion on https://github.com/OpenRefine/OpenRefine/issues/1674. Will amend my PR with tests tomorrow and update! Thank you for quick responses and encouragement!

File: main/tests/server/src/com/google/refine/tests/expr/functions/strings/SmartSplitTests.java [hunk 1]
```diff
@@ -1,15 +1,100 @@
...
+        Function function = ControlFunctionRegistry.getFunction(name);
+        if (function == null) {
+            throw new IllegalArgumentException("Unknown function " + name);
+        }
+        if (args == null) {
+            return function.call(bindings, new Object[0]);
+        } else {
+            return function.call(bindings, args);
+        }
+    }
```
Comment (wetneb) [inline main/tests/server/src/com/google/refine/tests/expr/functions/strings/SmartSplitTests.java]: Fantastic, thanks a lot! As a small note, I think `invoke` should not need to take the function name as parameter, as it is always `FUNCTION_NAME` in this class (and the method is private). But that is a cosmetic concern.

Comment (omkarnix) [reply main/tests/server/src/com/google/refine/tests/expr/functions/strings/SmartSplitTests.java]: Yeah. Actually I thought we should move the invoke method to some test util class or to probably RefineTest. Actually sorry I forgot to extend SmartSplitTest class by RefineTest. I think I should probably add that code right away.

Comment (wetneb) [reply main/tests/server/src/com/google/refine/tests/expr/functions/strings/SmartSplitTests.java]: @omkarnix moving to a test util class would make a lot of sense too.


The `RefineTest` class is used to create a test workspace and initialize various other things that are required when dealing with projects. For function tests it is not needed to extend it, as function registration is done statically as you have rightly noted.

Comment (omkarnix) [reply main/tests/server/src/com/google/refine/tests/expr/functions/strings/SmartSplitTests.java]: Ok. Thanks. I would certainly like to refactor it and move the invoke method to some util class. 
I noticed that travis CI build failed due to assertion failure in serializeProcessManager whereas appveyor build succeeded. Could it be a glitch while building on travis-ci? Can we trigger the build manually?

Comment (omkarnix) [reply main/tests/server/src/com/google/refine/tests/expr/functions/strings/SmartSplitTests.java]: > Fantastic, thanks a lot! As a small note, I think `invoke` should not need to take the function name as parameter, as it is always `FUNCTION_NAME` in this class (and the method is private). But that is a cosmetic concern.

@wetneb I agree. It was really not required to create constant for the local static value. So made the change suggested by yo .Also added one more misc change. Can you review it?

Comment (wetneb) [reply main/tests/server/src/com/google/refine/tests/expr/functions/strings/SmartSplitTests.java]: yes don't worry about unrelated test failures (although this one is curious). I can restart the build when it happens.

**Comment by omkarnix** [comment]
@wetneb Thank you very much for your support and encouragement on contribution. The community is very supportive. I would certainly try to contribute as much as possible!

**Pull request merged**